core/lib/zk_os_merkle_tree/src/hasher/nodes.rs

```diff
@@ -0,0 +1,183 @@
+//! Hashing for tree nodes.
+
+use std::{collections::HashMap, iter};
+
+use zksync_basic_types::H256;
+
+use crate::{
+    max_nibbles_for_internal_node,
+    types::{InternalNode, Root},
+    HashTree, TreeParams,
+};
+
+/// Internal hashes for a single `InternalNode`. Ordered by ascending depth `1..internal_node_depth`
+/// where `depth == 1` is just above child refs. I.e., the last entry contains 2 hashes (unless the internal node is incomplete),
```

Comment:
what do you mean by incomplete?

Reply:
Containing less child refs than the maximum possible amount (e.g., <16 for radix-16 amortization). I've tried to explain amortization in the readme.